From 769fa5f1728719b098c520a8a1e6416925466ebc Mon Sep 17 00:00:00 2001
From: Abhishek <apatgaonkar@habana.ai>
Date: Thu, 20 Jun 2024 14:01:02 +0300
Subject: [PATCH] [FSW-2704] fs1: support two manifests in data structures

In two manifest design, each manifest will have single
instance of ppboot and recovery images required. There
will be no support for tpm spi. To accommodate this design,
after image validation has passed rom will store only
one manifest header and relevant image descriptors
in persistent data boot_rom_manifest.
This persistent data is redundant if boot info is
enabled and rom will read this data from boot info
file.

Change-Id: Ie5070ef414bcb624981b1af31bc614fb5536fd6b
---

diff --git a/common/hl_boot_desc_rom.h b/common/hl_boot_desc_rom.h
index 1517104..f398307 100644
--- a/common/hl_boot_desc_rom.h
+++ b/common/hl_boot_desc_rom.h
@@ -18,16 +18,8 @@
 	HL_FW_MAX_RECOVERY_BOOT_PTRS,
 };
 
-enum hl_fw_boot_ptr_imgs {
-	HL_FW_BOOT_PTR_RECOVERY, /* RECOVERY Images */
-	HL_FW_BOOT_PTR_SPI, /* SPI FW Images */
-	HL_FW_BOOT_PTR_MAX_IMG,
-};
-
-struct hl_fw_boot_ptrs_ctx {
-	uint8_t boot_desc_flag;
-	uint32_t img_ptr_spi_offset;
-	struct hl_fw_desc_info desc_info;
+struct hl_fw_manifest_constants {
+	uint32_t manifest_spi_offset;
 	uint64_t dest_addr_dma;
 	uint32_t dest_addr_arc;
 };
diff --git a/include/boot_rom.h b/include/boot_rom.h
index 4a3532d..e1dbb9e 100644
--- a/include/boot_rom.h
+++ b/include/boot_rom.h
@@ -39,6 +39,12 @@
 #define STATIC_DIE_TYPE_IO				0x5
 #define STATIC_DIE_TYPE_COMP				0xA
 
+enum boot_rom_manifests {
+	BOOT_ROM_PRIMARY = 0,
+	BOOT_ROM_SECONDARY = 1,
+	BOOT_ROM_NUM_OF_MANIFESTS = 2
+};
+
 enum boot_rom_status {
 	BOOT_ROM_SUCCESS,
 	BOOT_ROM_FAIL,
@@ -49,7 +55,8 @@
 enum rom_internal_boot_desc_idx {
 	ROM_DESC_PPBOOT,
 	ROM_DESC_FAKE_FUSE,
-	ROM_DESC_TPM,
+	ROM_DESC_RECOVERY_0,
+	ROM_DESC_RECOVERY_1,
 	ROM_DESC_MISC_1,
 	ROM_DESC_MISC_2,
 	ROM_DESC_MISC_3,
@@ -65,6 +72,15 @@
 	TPM_EXEC_PCR_EXTEND_SEND = 3,
 	TPM_EXEC_PCR_EXTEND_RECV = 4,
 	TPM_EXEC_DONE = 5,
+};
+
+#define NUM_OF_MISC_DESCS	(ROM_DESC_NUM_MAX - ROM_DESC_MISC_1)
+
+
+struct hl_fw_manifest_ctx {
+	uint8_t boot_desc_flag;
+	struct hl_fw_desc_info desc_info;
+	struct hl_fw_img_desc spi_img_desc[ROM_DESC_NUM_MAX];
 };
 
 struct boot_rom_global_s {
@@ -90,8 +106,8 @@
 };
 
 struct boot_rom_manifest_s {
-	struct hl_fw_boot_ptrs_ctx boot_ptr_ctx[HL_FW_BOOT_PTR_MAX_IMG];
-	struct hl_fw_img_desc spi_img_desc[ROM_DESC_NUM_MAX];
+	struct hl_fw_boot_ptrs spi_manifest_hdr;
+	struct hl_fw_manifest_ctx manifest_ctx;
 };
 
 #define SYS_REF_CLOCK		50000000ULL
